lang_js/analyze/ast_js.ml

```diff
@@ -136,7 +136,7 @@ type special =
   | In | Delete 
   | Spread
   | Yield | YieldStar | Await
-  | Encaps of name option (* less: resolve? *)
+  | Encaps of bool (* if true, first arg of apply is the "tag" *)
```

Comment:
I could also imagine leveraging typing here:

```ocaml
| EncapsCall of encaps
| Encaps
```
where the latter is just a nullary variant.

Reply:
Actually it should be Encaps of expr option, to make the difference between
`foo ${x}` and tag`foo ${x}` but that would require the special type to be mutually recursive with expr, which I don't want. The motivation for special is actually to reduce the size of the cases in the expression, to simplify visitor/mappers and more generally any algorithm working on expressions, but at the same time I don't want to lose too much information and put everything in a call. For me Eval is special, as well as This/Super/... hence this type.
We lose indeed typing information by putting may things in special without any information about the arity of those special functions/values, but that's the tradeoff ... less typing security, but less cases to handle in most algorithms.